backend/src/utils/validation/isValidComplexity.ts

```diff
@@ -0,0 +1,6 @@
+import { UserRank } from '../../enums/users';
+
+const isValidComplexity = (complexity?: UserRank): boolean =>
+  Boolean(complexity && typeof complexity === 'number');
```

Comment:
not enough checks. complexity also should be positive and should contain only those values, that described in `UserRank` enum. That is really important for some other logic parts of our api